### ISSUE TITLE
readable game output

### ISSUE THREAD

**Issue by tissatussa** [cutechess/cutechess#662]
the CuteChess output log looks like this :

```
1. Bxh7+ {+0.96/24 20s} Kxh7 {-1.42/23 41s} 2. Qh5+ {+1.20/29 30s}
Kg8 {-0.24/18 7.6s} 3. O-O {+1.20/30 63s} Bxg5 {-2.01/25 83s}
4. fxg5 {+1.13/29 1.4s} Nxe5 {-1.77/22 23s} 5. Bc5 {+1.67/28 63s}
Na6 {-1.97/24 3.7s} 6. Bd4 {+1.78/29 54s} Ng6 {-0.44/23 24s}
[...]
```

but for readability i would be nice to display it like this :

```
1. Bxh7+ {+0.96/24 20s} Kxh7 {-1.42/23 41s}
2. Qh5+ {+1.20/29 30s} Kg8 {-0.24/18 7.6s}
3. O-O {+1.20/30 63s} Bxg5 {-2.01/25 83s}
4. fxg5 {+1.13/29 1.4s} Nxe5 {-1.77/22 23s}
5. Bc5 {+1.67/28 63s} Na6 {-1.97/24 3.7s}
6. Bd4 {+1.78/29 54s} Ng6 {-0.44/23 24s}
[...]
```

or even like this :

```
1. Bxh7+ {+0.96/24 20s}  Kxh7 {-1.42/23 41s}
2. Qh5+  {+1.20/29 30s}  Kg8  {-0.24/18 7.6s}
3. O-O   {+1.20/30 63s}  Bxg5 {-2.01/25 83s}
4. fxg5  {+1.13/29 1.4s} Nxe5 {-1.77/22 23s}
5. Bc5   {+1.67/28 63s}  Na6  {-1.97/24 3.7s}
6. Bd4   {+1.78/29 54s}  Ng6  {-0.44/23 24s}
[...]
```

**Comment by alwey** [comment]
The sample PGN output from the PGN specification document looks like this:

```
[Event "F/S Return Match"] 
[Site "Belgrade, Serbia JUG"] 
[Date "1992.11.04"] 
[Round "29"] 
[White "Fischer, Robert J."]
[Black "Spassky, Boris V."] 
[Result "1/2-1/2"] 

1. e4 e5 2. Nf3 Nc6 3. Bb5 a6 4. Ba4 Nf6 5. O-O Be7 6. Re1 b5 7. Bb3 d6 8. c3
O-O 9. h3 Nb8 10. d4 Nbd7 11. c4 c6 12. cxb5 axb5 13. Nc3 Bb7 14. Bg5 b4 15.
Nb1 h6 16. Bh4 c5 17. dxe5 Nxe4 18. Bxe7 Qxe7 19. exd6 Qf6 20. Nbd2 Nxd6 21.
Nc4 Nxc4 22. Bxc4 Nb6 23. Ne5 Rae8 24. Bxf7+ Rxf7 25. Nxf7 Rxe1+ 26. Qxe1 Kxf7
27. Qe3 Qg5 28. Qxg5 hxg5 29. b3 Ke6 30. a3 Kd6 31. axb4 cxb4 32. Ra5 Nd5 33.
f3 Bc8 34. Kf2 Bf5 35. Ra7 g6 36. Ra6+ Kc5 37. Ke1 Nf4 38. g3 Nxh3 39. Kd2 Kb5
40. Rd6 Kc5 41. Ra6 Nf2 42. g4 Bd3 43. Re6 1/2-1/2
```

I think many PGN users prefer a condensed presentation in order to have major portions of a game visible without paging or scrolling.

PGN data can be reformatted by dedicated tools.

The cutechess GUI lists the moves of the current game with a new line for every full move. You can switch the comments off and on by right-clicking the list of moves and selecting "Toggle Comments" or by using the key "c".

**Comment by alwey** [comment]
![l1](https://user-images.githubusercontent.com/6425738/152200226-fbe2e6e9-a106-4242-91cc-172f96eab078.png)
_Move list with comments_


![l2](https://user-images.githubusercontent.com/6425738/152200264-21d62815-e7e4-4424-b856-2cfb91fe1836.png)
_Formatted move list without comments_

**Comment by tissatussa** [comment]
> You can switch the comments off and on by right-clicking the list of moves and selecting "Toggle Comments" or by using the key "c".

No, my CuteChess v1.2.0 gives no such option in the right-mouse-menu -- i'm on Linux (Devuan Chimaera 4.0 - a Debian derivative).

![cutechess_rightmousemenu](https://user-images.githubusercontent.com/1109281/152205062-6fd7e18e-7b76-4b95-9cf8-8e44993a022d.png)

**Comment by tissatussa** [comment]
> Do you need the list on screen or on file?

To convert the PGN file notation into such formatted list, i can write a (python) script myself, but it would be nice both screen and file output can be like this.

**Comment by alwey** [comment]
I think about copying the listing and using the game's PGN header. Adding a context menu item either  "Save to File ..." or "Save to Clipboard"?

**Comment by tissatussa** [comment]
> ..context menu item either "Save to File" or "Save to Clipboard"..

Both can be handy, but i think Save to Clipboard is more handy : the user can then save it to file. It would really be nice if you implement this, i think many users will like this feature. I often look at the thinking times. My first textt of this Issue mentions two "proper" list displays, the second being most readable. Btw. not all engines output an eval value at each move, so aligning all move lines will be a bit tricky ..

**Comment by alwey** [comment]
> > context menu item either "Save to File" or "Save to Clipboard"

> Both can be handy, but i think Save to Clipboard is more handy : the user can then save it to file. It would really be nice if you implement this, i think many users will like this feature.

I would go with the clipboard option. Cheaper and more flexible

> Btw. not all engines output an eval value at each move, so aligning all move lines will be a bit tricky ..

Exactly, and additionally I fear we are expected to some day support more than the Xboard-Cutechhess style comments like `{evaluation/depth movetime}` but also `(pondermove)` , `(variant moves)` and extended commands like `%clk` `%emt` or TCEC comment format.

**Comment by tissatussa** [comment]
if it's all python, maybe i can help you with that .. i think building a filter to output a nice list will not be hard .. but i have no clue how to use GitHub .. i never did a PR or so .. i can work in terminal though.

**Comment by alwey** [comment]
Thank you. Indeed, I will have a C++-patch ready on Saturday. 

Nevertheless, I think that scripting support is welcome. You will have to create a fork of the cutechess repository on github (find the buttton close to the top of this page). In a second step you visit your new forked cutechess repository page and then clone it onto your computer (press the button Code on your repository page and you will see the URI), e.g. `https://github.com/tissatussa/cutechess.git.` You have to install git on your computer, reserve a directory for repositories, go to that directory and issue the command `git clone https://github.com/tissatussa/cutechess.git`. Then the cutechess code is transferred onto your computer where you can work on it. Use a new branch for work, e.g. ` git checkout master -b mynewbranch`

If you come up with something, you have to `push` the branch with your new code to your cutechess repository on github. If all is good, then, finally, you can create a pull request for cutechess/cutechess. There are Help pages to assist you.
